engine/src/mbliphoneapp.mm

```diff
@@ -505,7 +520,19 @@ - (void)application:(UIApplication *)p_application didReceiveRemoteNotification:
     UIApplicationState t_state = [p_application applicationState];
     MCAutoStringRef t_mc_push_notification_text;
     NSString *t_reminder_text = [p_dictionary objectForKey:@"payload"];
```

Comment:
This isn't strictly correct if the return of objectForKey: is NSNull and not NSString. It is better to make this explicit:
```
id t_reminder_text_value = [p_notification.userInfo objectForKey:@"payload"];
NSString *t_reminder_text;
if ([t_reminder_text_value isKindOfClass: [NSString class]])
  t_reminder_text = t_reminder_text_value;
else
  t_reminder_text = nil
```